### ISSUE TITLE
SystemJS compile issue with file names starting with "user*"

### ISSUE THREAD

**Issue by ricardobrandao** [systemjs/builder#740]
I've hit this weird bug where the systemjs builder throws an **invalid token** exception when parsing files starting ```u*.js``` during compile (i.e. builder.compile).

I've created a [repository ](https://github.com/ricardobrandao/systemjs-issue) that shows this issue.

I'm using jspm v0.16.48.

**Comment by aluanhaddad** [comment]
It looks like this happens in normalize whenever the name of the requested module begins with `u`.

**Comment by guybedford** [comment]
@aluanhaddad I just tested the above repo and it doesn't give me any issues. Perhaps this is a windows thing?

**Comment by ricardobrandao** [comment]
@guybedford it looks that this only happens on windows - just tested on mac and it worked just fine. 

On windows it throws the following error:
```
[09:44:12] 'build-js' errored after 560 ms
[09:44:12] Error on dependency parsing for users\user-model.js at file:///c:/repos/systemjs-issue/src/users\user-model.js
        (SystemJS) Unexpected token ILLEGAL
        SyntaxError: Unexpected token ILLEGAL
            at Object.exports.runInThisContext (vm.js:53:16)
            at SystemJSNodeLoader.loader.instantiate (c:\repos\systemjs-issue\node_modules\jspm\node_modules\systemjs-builder\compilers\amd.js:292:30)
            at SystemJSNodeLoader.loader.instantiate (c:\repos\systemjs-issue\node_modules\jspm\node_modules\systemjs-builder\compilers\esm.js:54:32)
            at c:\repos\systemjs-issue\node_modules\jspm\node_modules\systemjs-builder\lib\trace.js:533:25
            at tryCatcher (c:\repos\systemjs-issue\node_modules\jspm\node_modules\systemjs-builder\node_modules\bluebird\js\release\util.js:16:23)
            at Promise._settlePromiseFromHandler (c:\repos\systemjs-issue\node_modules\jspm\node_modules\systemjs-builder\node_modules\bluebird\js\release\promise.js:510:31)
            at Promise._settlePromise (c:\repos\systemjs-issue\node_modules\jspm\node_modules\systemjs-builder\node_modules\bluebird\js\release\promise.js:567:18)
            at Promise._settlePromise0 (c:\repos\systemjs-issue\node_modules\jspm\node_modules\systemjs-builder\node_modules\bluebird\js\release\promise.js:612:10)
            at Promise._settlePromises (c:\repos\systemjs-issue\node_modules\jspm\node_modules\systemjs-builder\node_modules\bluebird\js\release\promise.js:691:18)
            at Async._drainQueue (c:\repos\systemjs-issue\node_modules\jspm\node_modules\systemjs-builder\node_modules\bluebird\js\release\async.js:133:16)
        Evaluating c:/repos/systemjs-issue/src/users\user-model.js
```

**Comment by guybedford** [comment]
@ricardobrandao thanks it looks like the issue is the bad URL conversion of `file:///c:/repos/systemjs-issue/src/users\user-model.js`, which would be interpreting `\u` as a control character in the output string.

We probably just need to add an extra URL sanitization somewhere then to fix this up.

**Comment by aluanhaddad** [comment]
Yeah, this seems to happens in `coreResolve`. There are a lot of checks that look for the literal `'/'` and would miss a `'\'`.

**Comment by guybedford** [comment]
I've added a fix into the URL resolver at SystemJS master at https://github.com/systemjs/systemjs/commit/0008aa5efef5a0a1a7266212957c9448bd2988fe for the next release.

**Comment by guybedford** [comment]
This should be fixed in the latest 0.15 and 0.16.
